src/main/java/eu/donals/UserInterface.java

```diff
@@ -0,0 +1,343 @@
+package eu.donals;
+
```

Comment:
I think the nature of a project like this one sort of forces you to make large files. You managed to make them very maintainable and legible which is impressive. You do use a lot of magic numbers, but again, I wouldn't know how to work around them in this case.